pkg/segment/structs/segstructs.go

```diff
@@ -378,33 +379,34 @@ type QueryCount struct {
-	TotalRRCCount             uint64
-	MeasureFunctions          []string        `json:"measureFunctions,omitempty"`
-	MeasureResults            []*BucketHolder `json:"measure,omitempty"`
-	GroupByCols               []string        `json:"groupByCols,omitempty"`
-	Qtype                     string          `json:"qtype,omitempty"`
-	BucketCount               int             `json:"bucketCount,omitempty"`
-	PerformAggsOnRecs         bool            // if true, perform aggregations on records that are returned from rrcreader.go
-	RecsAggsType              PipeCommandType // To determine Whether it is GroupByType or MeasureAggsType
-	GroupByRequest            *GroupByRequest
-	MeasureOperations         []*MeasureAggregator
-	NextQueryAgg              *QueryAggregators
-	RecsAggsBlockResults      interface{}              // Evaluates to *blockresults.BlockResults
-	RecsAggsColumnKeysMap     map[string][]interface{} // map of column name to column keys for GroupBy Recs
-	RecsAggsProcessedSegments uint64
-	RecsRunningSegStats       []*SegStats
-	TransactionEventRecords   map[string]map[string]interface{}
-	TransactionsProcessed     map[string]map[string]interface{}
-	ColumnsOrder              map[string]int
-	RawSearchFinished         bool
-	CurrentSearchResultCount  int
+	AllRecords                  []*utils.RecordResultContainer
+	ErrList                     []error
+	Histogram                   map[string]*AggregationResult
+	TotalResults                *QueryCount
+	VectorResultValue           float64
+	RenameColumns               map[string]string
+	SegEncToKey                 map[uint16]string
+	TotalRRCCount               uint64
+	MeasureFunctions            []string        `json:"measureFunctions,omitempty"`
+	MeasureResults              []*BucketHolder `json:"measure,omitempty"`
+	GroupByCols                 []string        `json:"groupByCols,omitempty"`
+	Qtype                       string          `json:"qtype,omitempty"`
+	BucketCount                 int             `json:"bucketCount,omitempty"`
+	PerformAggsOnRecs           bool            // if true, perform aggregations on records that are returned from rrcreader.go
+	RecsAggsType                PipeCommandType // To determine Whether it is GroupByType or MeasureAggsType
+	GroupByRequest              *GroupByRequest
+	MeasureOperations           []*MeasureAggregator
+	NextQueryAgg                *QueryAggregators
+	RecsAggsBlockResults        interface{}              // Evaluates to *blockresults.BlockResults
+	RecsAggsColumnKeysMap       map[string][]interface{} // map of column name to column keys for GroupBy Recs
+	RecsAggsProcessedSegments   uint64
+	RecsRunningSegStats         []*SegStats
+	TransactionEventRecords     map[string]map[string]interface{}
+	TransactionsProcessed       map[string]map[string]interface{}
+	ColumnsOrder                map[string]int
+	RawSearchFinished           bool
+	CurrentSearchResultCount    int
+	AllSearchColumnsByTimeRange map[string]bool
```

Comment:
nit: this should be a `map[string]struct{}` because the `bool` is never `false`, right?

Reply:
Yes, that might be case but I want to keep the type same as `finalCols`, so that I can use it in functions like: `applyColNameTransform` in `rrcreader.go` and probably in future with some other functions where I want to send this value instead of `finalCols`

Reply:
Ok you can keep this is if you want, or change it if you want